athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/ConnectorStackAttributesProvider.java

```diff
@@ -0,0 +1,109 @@
+ * #L%
+ */
+package com.amazonaws.athena.connector.integ;
+
+import com.amazonaws.athena.connector.integ.data.ConnectorPackagingAttributes;
+import com.amazonaws.athena.connector.integ.data.ConnectorStackAttributes;
+import com.amazonaws.athena.connector.integ.data.ConnectorVpcAttributes;
+import com.amazonaws.athena.connector.integ.providers.ConnectorPackagingAttributesProvider;
+import com.amazonaws.athena.connector.integ.providers.ConnectorVpcAttributesProvider;
+import software.amazon.awscdk.core.Construct;
+import software.amazon.awscdk.services.iam.PolicyDocument;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Responsible for providing the Connector's stack attributes used in creating the Connector's stack.
+ */
+public class ConnectorStackAttributesProvider
+{
+    private static final String LAMBDA_SPILL_BUCKET_TAG = "spill_bucket";
+    private static final String LAMBDA_SPILL_PREFIX_TAG = "spill_prefix";
+    private static final String LAMBDA_DISABLE_SPILL_ENCRYPTION_TAG = "disable_spill_encryption";
+
+    private final Construct scope;
+    private final String id;
+    private final String lambdaFunctionName;
+    private final Map<String, Object> testConfig;
+    private final Optional<PolicyDocument> connectorAccessPolicy;
+    private final Map<String, String> environmentVariables;
+    private final ConnectorPackagingAttributes connectorPackagingAttributes;
+    private final Optional<ConnectorVpcAttributes> connectorVpcAttributes;
+
+    protected ConnectorStackAttributesProvider(final Construct scope, final String id, final String lambdaFunctionName,
+                                               final Map<String, Object> testConfig,
+                                               final Optional<PolicyDocument> connectorAccessPolicy,
+                                               final Map<String, String> environmentVariables)
+    {
+        this.scope = scope;
+        this.id = id;
+        this.lambdaFunctionName = lambdaFunctionName;
+        this.testConfig = testConfig;
+        this.connectorAccessPolicy = connectorAccessPolicy;
+        this.environmentVariables = environmentVariables;
+        this.connectorPackagingAttributes = ConnectorPackagingAttributesProvider.getAttributes();
+        this.connectorVpcAttributes = ConnectorVpcAttributesProvider.getAttributes(testConfig);
+
+        setUpEnvironmentVars();
+    }
+
+    /**
+     * Sets defaults for environment variables (spill_bucket, spill_prefix, disable_spill_encryption) if not provided
+     * by connector.
+     * @return A Map containing the environment variables key-value pairs.
+     * @throws RuntimeException The spill_bucket is neither specified in the environment vars nor the test config file.
+     */
+    private void setUpEnvironmentVars()
+            throws RuntimeException
+    {
+        // Check for missing spill_bucket
+        if (!environmentVariables.containsKey(LAMBDA_SPILL_BUCKET_TAG)) {
```

Comment:
remind me why we are using environment variables for anything instead of only using the config file? seems confusing to have a mix.

Reply:
moved to config file.